javaslang-benchmark/src/test/java/javaslang/idiom/TryBenchmark.java

```diff
@@ -0,0 +1,111 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import javaslang.control.Try;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.util.Random;
+
+import static javaslang.API.Case;
+import static javaslang.API.Match;
+import static javaslang.Predicates.instanceOf;
+
+/**
+ * Benchmark for Try vs try/catch.
+ */
+public class TryBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            TryThrowsCatchBenchmark.class,
+            TryNoThrowsCatchBenchmark.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+
+    @State(Scope.Benchmark)
+    public static class Base {
+
+        int random;
+
+        Integer throwsException() {
+            if ((random ^ random) == 0) throw new RuntimeException();
+            return 1;
+        }
+
+        Integer doesntThrow() {
+            if ((random ^ random) != 0) throw new RuntimeException();
+            return 0;
+        }
+
+        @Setup
+        public void setup() {
+            random = new Random().nextInt();
+        }
+    }
+
+    public static class TryThrowsCatchBenchmark extends Base {
+
+        @Benchmark
+        public Object javaslang_try_actual_exception() {
+            Integer res = Try.of(this::throwsException)
+                    .recover(x -> Match(x).of(Case(instanceOf(RuntimeException.class), () -> 1)))
```

Comment:
looking at this we should add a method `Try.recover(Class, Function)`:

```java
// recovers this Try with f if this is a failure of type X, otherwise returns this
default <X extends Throwable> Try<T> recover(Class<X> exception, Function<? super X, ? extends T> f) {
    if (isFailure()) {
        final Throwable cause = getCause();
        if (exception.isAssignableFrom(cause.getClass())) {
            return Try.of(() -> f.apply(cause));
        }
    }
    return this;
}
```

in order to get a way around the Match expression. Then we have

```java
Try.of(this::throwsException).recover(RuntimeException.class, x -> 1);
```